portable/ThirdParty/GCC/Posix/utils/wait_for_event.c

```diff
@@ -21,11 +21,17 @@ struct event * event_create()
     return ev;
 }
 
-void event_delete( struct event * ev )
+void event_delete( struct event ** ev_ptr )
 {
-    pthread_mutex_destroy( &ev->mutex );
-    pthread_cond_destroy( &ev->cond );
-    free( ev );
+    struct event * ev = *ev_ptr;
+
+    if ( ev )
+    {
+        pthread_mutex_destroy( &ev->mutex );
+        pthread_cond_destroy( &ev->cond );
+        free( ev );
+        *ev_ptr = NULL;
+    }
```

Comment:
This would hide the doube deletion of an even if called twice.
We would like it to fail as soon as possible, as it could make a design issue not apparent by trying to delete a task twice.
This is similar to how free works, if you call free twice on a pointer the second time it would crash instead of checking the validity of that pointer

Reply:
Yes, you are correct. This will hide the double deletion. I added this check to be able to work with the fix in PR #181 when `event_delete` where called from two places. But now this is not needed after refining the fix of PR #181 by deleting the condition variable in the correct place (When a thread is canceled).